### PR TITLE
chore: Only verify Jdk 9 classes on jdk 11.

### PR DESCRIPTION
**Pull request by He-Pin**
Motivation:
Polish the script to test across all scala versions on JDK 11.

eg:

>Starting verification for with file: stream/target/scala-cli/VerifyJDK9Classes-2.13.sc.
Verification successful for stream/target/scala-cli/VerifyJDK9Classes-2.13.sc.
All verifications successful.

demo: https://github.com/apache/incubator-pekko/actions/runs/7694919153/job/20966708057?pr=1078

### CODE REVIEW

File: project/VerifyJDK9Classes.scala [hunk 1]
```diff
@@ -22,24 +22,23 @@ object VerifyJDK9Classes {
   lazy val settings: Seq[Setting[_]] = inConfig(Compile) {
     Seq {
       sourceGenerators += {
-        generateAndWriteScalaCLIScript(
-          target,
-          _ / "scala-cli" / "VerifyJDK9Classes.sc")
+        generateAndWriteScalaCLIScript(target)
```
Comment (Roiocam) [inline project/VerifyJDK9Classes.scala:25]: More readable!

File: .github/workflows/nightly-builds.yml [hunk 2]
```diff
@@ -181,19 +181,15 @@ jobs:
...
+        if: ${{ matrix.javaVersion == 11 }}
         run: |-
-          scala-cli --version
-          echo "Starting verification with Scala-CLI"
-          scala-cli stream/target/scala-cli/VerifyJDK9Classes.sc && echo "Verification successful" || (
-          echo "Error when VerifyJDK9Classes"
-          exit 1
-          )
+          chmod +x .github/workflows/verify-jdk9-classes.sh
+          .github/workflows/verify-jdk9-classes.sh
```
Comment (Roiocam) [inline .github/workflows/nightly-builds.yml:195]:
```suggestion
          sh github/workflows/verify-jdk9-classes.sh
```

File: .github/workflows/verify-jdk9-classes.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+echo "Checking Scala CLI version..."
+scala-cli --version
+
+echo "Starting verification with Scala-CLI..."
+
+for file in stream/target/scala-cli/*.sc; do
```
Comment (Roiocam) [inline .github/workflows/verify-jdk9-classes.sh:8]: Cool, has expandable capability.

**Review by Roiocam — APPROVED** [review]
LGTM, one picky issue.

File: .github/workflows/nightly-builds.yml (hunk 2 above)
Comment (He-Pin) [reply .github/workflows/nightly-builds.yml:195]: Let's keep it for now, otherwise I have to revaluation it.

**Review by jxnu-liguobin — APPROVED** [review]

**Pull request merged**